cloud-third-party-controller/views/customui.html

```diff
@@ -0,0 +1,526 @@
+        url: "/golive/",
+        method: 'POST',
+        contentType: 'application/json',
+        processData: false,
+        success: function(data) {
+          // TODO: make this update based on the actual outputs statuses, rather than toggling and assuming all is working properly
+          if(isStreaming) {
+            if($("#go_live_button").hasClass("select-button-active")) {
+              $("#go_live_button").removeClass("select-button-active");
+              $("#go_live_button").addClass("select-button-ready");
+              $("#stream_button_text").text("Ready!");
+            }
+            isStreaming = false;
+          } else {
+            $("#go_live_button").removeClass("select-button-ready");
+            $("#go_live_button").addClass("select-button-active");
+            $("#stream_button_text").text("STREAMING\nLIVE");
+            isStreaming = true;
+          }
+        },
+        error: function(error) {
+          alert("An error occurred:\n\n" + JSON.stringify(error));
+        }
+      });
+    }
+
+    function submitPAT(token) {
+      var orgLabel = document.getElementById("org_label");
+      orgLabel.textContent = "Retrieving Organizations...";
+
+      $.ajax({
+        url: "/submitPAT/" + token,
+        method: 'POST',
+        contentType: 'application/json',
+        success: function(data) {
+          var orgSelect = document.getElementById("orgDropdown");
+          var orgs = JSON.parse(data)["orgs"];
+          for (var org in orgs) {
+            var newOrg = document.createElement("option");
+            newOrg.text = orgs[org].org_name;
+            newOrg.value = orgs[org].org_guid;
+            orgSelect.add(newOrg);
+          }
+          if (!orgSelect.options[1]) {
+            alert("No organizations found, check the entered Personal Access Token and that the associated account is part of an organization");
+            orgSelect.disabled = true;
+          } else {
+            orgSelect.disabled = false;
+          }
+          orgLabel.textContent = "LiveEdge Cloud Organizations";
+        },
+        error: function(error) {
+          alert("An error occurred:\n\n" + JSON.stringify(error));
+        }
+      });
+    }
+
+    function getDevices(token, org_guid) {
+      var deviceLabel = document.getElementById("device_label");
+      deviceLabel.textContent = "Retrieving Devices...";
+
+      $.ajax({
+        url: "/getDevices/",
+        method: 'POST',
+        data: JSON.stringify({token: token, org_guid: org_guid}),
+        dataType: "json",
+        contentType: 'application/json',
+        success: function(data) {
+          var deviceSelect = document.getElementById("deviceDropdown");
+          var deviceDiv = document.getElementById("device_select");
+          var devices = data["devices"];
+          for (var device in devices) {
+            var newDevice = document.createElement("option");
+            newDevice.text = devices[device].serial_number;
```

Comment:
Maybe add a step with a call to `https://api.videoncloud.com/v1/devices/{device_guid}/state` in order to get the `device_name` and `online` status of each device.

Reply:
My concern here is that with /devices not being designed to return device_name and /state not actually returning device_name, a call would have to be made for the system shadow, then parse out the device name from that, it might cause the loading of the device list to take a bit longer, especially with large fleets (maybe the case for either /state or /shadows anyway).

I can give it a try regardless, just wanted to document my thoughts

Reply:
Managed to do this asynchronously, devices will default to serial, but will be updated with assigned name after the initial get of the device list.